docs/week-3.md

```diff
@@ -0,0 +1,523 @@
+props is to compare them to how functions work. For example, let us print a message to the console.
+
+```
+function printHi() {
+
+    console.log("hi");
+
+}
+```
+
+We successfully printed "hi" to the console! Now what if we wanted to print "hey" to the console. We would have
+to make a new function because `printHi()` only prints "hi". We would then have to make `printHey()`.
+
+```
+function printHey() {
+
+    console.log("hey");
+
+}
+```
+
+We have now printed "hi" and "hey", but it is very repetitive and not efficient. If we wanted to print another
+message we would have to make a new function OR we can just add a parameter to a function to print whatever we
+want to the console and we would not need to make a new function each time. All we would need to do is pass in
+our message as a parameter like so.
+
+```
+function printMessage(message) {
+
+    console.log("Here is your note: " + message);
+
+}
+```
+
+Now we can reuse the function whenever we want to print something to the console! 
+
+### Using Props
+This same logic can be applied when we use props. Let us revisit the `LuckyNum` component again.
+
+```
+import React from 'react'; 
+
+const LuckyNum = () => {
+
+    let myLuckyNumber = 7;
+
+    return(
+        <div>
+            <h1>My Lucky Number is {myLuckyNumber}</h1>
+        </div>
+    );
+}
+
+export default LuckyNum;
+```
+
+Right now this component only displays John's lucky number. In order to make this component more reusable so that
+Doug can use it, we must add props. We add props as a parameter of the function component.
+
+```
+import React from 'react'; 
+
+const LuckyNum = (props) => {
+
+    let myLuckyNumber = 7;
+
+    return(
+        <div>
+            <h1>My Lucky Number is {myLuckyNumber}</h1>
+        </div>
+    );
+}
+
+export default LuckyNum;
+```
+
+Now that we have access to `props` we can simply display the luckyNum field of the prop object.
+
+```
+import React from 'react'; 
+
+const LuckyNum = (props) => (
+
+    <div>
+        <h1>My Lucky Number is {props.luckyNumber}</h1>
+    </div>
+
+)
+
+export default LuckyNum;
+```
+Our component is now reuseable!! We now can use it on Doug's profile like so.
+
+```
+import React from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is Doug! My favorite color is purple.</h1>
+        <LuckyNum luckyNum={1111}/>
+    </div>
+);
+
+export default Profile;
+```
+
+`props` is an object that React makes for each component. When you pass parameters into component declaration 
+like we did with `LuckNum` React adds the parameter as a field of the object `props`. In order to get the field
+of the object we simply access the object's field like we would any other object. 
+
+> We can make the `Profile` component more reusable with props. Try now to add props to make `Profile` more
```

Comment:
Try to add props to make `Profile` more generic yourself right now. 

or something like that